src/Workspaces/VisualBasic/Portable/Simplification/Reducers/VisualBasicInferredMemberNameReducer.vb

```diff
@@ -62,5 +70,40 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Simplification
 
             Return True
         End Function
+
+        ' An explicit name cannot be removed if some other position would produce it as inferred name
+        Private Shared Function RemovalCausesAmbiguity(arguments As SeparatedSyntaxList(Of SimpleArgumentSyntax), toRemove As SimpleArgumentSyntax) As Boolean
+            Dim name = toRemove.NameColonEquals.Name.Identifier.ValueText
+            For Each argument In arguments
+
+                If argument Is toRemove Then
+                    Continue For
+                End If
+
+                If argument.NameColonEquals Is Nothing AndAlso CaseInsensitiveComparison.Equals(argument.Expression.TryGetInferredMemberName(), name) Then
+                    Return True
+                End If
+            Next
+
+            Return False
+        End Function
+
+        ' An explicit name cannot be removed if some other position would produce it as inferred name
+        Private Shared Function RemovalCausesAmbiguity(initializers As SeparatedSyntaxList(Of FieldInitializerSyntax), toRemove As NamedFieldInitializerSyntax) As Boolean
+            Dim name = toRemove.Name.Identifier.ValueText
+            For Each initializer In initializers
+
+                If initializer Is toRemove Then
+                    Continue For
+                End If
+
+                Dim inferredInitializer = TryCast(initializer, InferredFieldInitializerSyntax)
+                If inferredInitializer IsNot Nothing AndAlso CaseInsensitiveComparison.Equals(inferredInitializer.Expression.TryGetInferredMemberName(), name) Then
```

Comment:
bummer that we can't share more coe here between C# and VB.  but NBD.

Reply:
Yup. I did give it a shot, but it ended with more convoluted code.